### ISSUE TITLE
Move From CSS to SCSS

### ISSUE THREAD

**Issue by williamjallen** [Submitty/Submitty#6425]
**What problem are you trying to solve with Submitty**
Submitty currently uses pure CSS to style the website.  It is common to see classes like `class_name` and `class-name` refer to two different problems in two completely different places and to encounter naming conflicts due to poorly-named classes.

**Describe the way you'd like to solve this problem**

I propose that Submitty switches to [SCSS](https://sass-lang.com/guide) instead of CSS moving forward.  There are many benefits of using SCSS and few downsides.  A few notable things that would be useful in Submitty are:

- Nesting makes scoping easy and gives CSS an obvious hierarchy
- Math operators for calculating fixed units 
- Cross-browser support using something like [Bourbon](https://github.com/thoughtbot/bourbon)
    - Automatically adds `moz-` and `webkit-` prefixes and does other things to maximize browser compatibility
- @includes add the ability to include SCSS files which reduces duplication and improves modularity
- Comments -- // comments are allowed in SCSS; comments are always good
- Minification the SCSS preprocessor outputs minified CSS bundles which reduces the resulting file size by 25-30%.

SCSS would be handled through NPM and SCSS can include regular CSS code so the total changes would be minor and the switchover could be as gradual as necessary.

**Comment by MasterOdin** [comment]
> It is common to see classes like class_name and class-name refer to two different problems in two completely different places and to encounter naming conflicts due to poorly-named classes.

It's not clear to me how a move to SCSS would solve either of these problems? SCSS would still allow people to mix `class_name` and `class-name` freely, and the latter is more an issue of naming and discoverability?

**Comment by williamjallen** [comment]
No, it wouldn't solve the issue directly.  Developers could still do that.  The benefit here is that it is easy to have some form of scoping by nesting the SCSS.  If you have page `a` and page `b`, you can wrap the CSS for each page in a wrapper such as the following:
```
a {
    div1 {
        width: 0;
        height: 1em;
    }
}

b {
    div1 {
        width: 10em;
        height: 20em;
    }
}
```

In this case, page `a` doesn't interfere with page `b` in any way and the developers of both pages are free to name their classes as they please.  Too much nesting is poor coding style but some amount of nesting such as the above helps readability and maintainability considerably.

**Comment by BenKelly99** [comment]
Maybe my CSS knowledge is failing but isn't that possible in normal CSS? Specific that only divs within an element of a specific class name?

**Comment by MasterOdin** [comment]
You can do this with regular CSS, but it's less nice to look at, especially with deeper levels of nesting, and having multiple items per nest.
